includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -193,6 +193,14 @@ public function payment_scripts() {
 			return;
 		}
 
+		if ( is_product() && ! WC_Stripe_Helper::should_load_scripts_on_product_page() ) {
+			return;
+		}
+
+		if ( is_cart() && ! WC_Stripe_Helper::should_load_scripts_on_cart_page() ) {
+			return;
+		}
```

Comment:
I actually don't think this is necessary. Do we run this gateway on the product and cart pages? I feel like we fall back to the base gateway class there. Haven't tested it though. Need to look into it.

Reply:
Based on testing: the `is_cart()` check is required, but the `is_product()` check is not. I've removed the `is_product()` check in 9bc82b5e8368ed31006eb4ded95555e36df24615.